File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from math import ceil

def solution(progresses, speeds):
    days=[]
    for i,j in zip(progresses,speeds):
        days.append(ceil((100-i)/j))
    answer=[]    
    print(days)
    stack=[days.pop(0)]
    while days:
        if stack[0] >= days[0]:
            stack.append(days.pop(0))
        else:
            answer.append(len(stack))
            stack=[days.pop(0)]
    answer.append(len(stack))
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const n = progresses.length
    
    const dueDates = []
    for(let i=0; i<n; i++){
        dueDates.push(Math.ceil((100-progresses[i])/speeds[i]))
    }
    
    
    let stack = []
    
    stack.push(dueDates.shift())
    
    const result = []
    while(dueDates.length){
        
        if(stack[0] >= dueDates[0]){
            stack.push(dueDates.shift())
        } else {
            result.push(stack.length)
            stack = [dueDates.shift()]
        }
        
    }
    
    result.push(stack.length)
    
    return result
}